docs/Setting-up-monorepo.md

```diff
@@ -0,0 +1,356 @@
+# Setting up a yarn monorepo with Ignite framework (React Native)
+
+In this tutorial, we'll guide you through the process of setting up a yarn monorepo for your React Native projects using the Ignite framework. We'll start by setting up the monorepo structure, create a React Native app using Ignite, a second web app plain vainilla Javascript, add a shared form-validator utility, and finally integrate this utility into both apps.
+
+## Prerequisites
+
+Before you start, ensure you have the following installed on your machine:
+
+- [Node.js](https://nodejs.org/en) (version 18 or later)
+- [Yarn](https://classic.yarnpkg.com/en/)
+
+## Use case
+
+In a monorepo setup, multiple apps often share common functionality. For example, a mobile (React Native) and a web app (React). This tutorial demonstrates how to set up and use shared utilities within a monorepo. For instance, if you have several apps that need to validate user inputs like email addresses, passwords, or text fields, you can create a single validation utility that can be reused across all your apps.
+
+By centralizing these utilities, you reduce code duplication and simplify maintenance, ensuring that any updates or bug fixes are immediately available to all your apps.
+
+In this setup, we’ll create a React Native app, a React app and a form-validator utility that includes functions to validate email addresses, passwords, and generic text fields. This utility will then be integrated to both apps.
+
+## Step 1: Setting up the monorepo
+
+First, follow the [Expo documentation on setting up monorepos](https://docs.expo.dev/guides/monorepos/) to initialize your own monorepo. This will include setting up your `packages/` and `apps/` directories and configuring Yarn workspaces.
```

Comment:
@jamonholmgren that's a surprising question to me in this context. @frankcalise's comment on the CNG default issue is the only effective difference in using Ignite with "bare" is that the android and iOS directories won't be gitignored: https://github.com/infinitered/ignite/issues/2638.

So to me in effect that sounds like everyone using Ignite is using Expo (in the ways in which it matters for this context, which would would be using Expo libraries and Expo's metro config.)

Reply:
@lindboe You're correct, technically, but we will still get this question and should address it here, IMO.

Reply:
As an example (please don't just copy & paste this, I've given it very little thought 😅 ):

> **What if I'm not using Expo?** Generally speaking, it shouldn't make any significant difference for this guide how you're using Expo (or even if you're using any Expo libraries at all). You can still follow the Expo monorepo guide up to the point before creating the app itself.

Reply:
BTW that Expo monorepo guide is very close to what I had in mind. It's very good, and we can draw inspiration from it (including all the little ways they educate and give hints along the way.)

![CleanShot 2024-08-13 at 16 12 37@2x](https://github.com/user-attachments/assets/142ba3c0-4632-415c-80e1-20b69e29646e)

Reply:
Okay thanks, that helps me understand what you're looking for 👍 

I think the main thing to call out would be that `expo/metro-config` has automatic monorepo support, as the guide notes:

> Expo's Metro config has monorepo support for bun, npm, and yarn.

Reply:
I'm pretty sure we have non-Expo (in any way) projects that are in monorepos, and could provide guidance for those as well; this could be a todo for later though.

Reply:
> I'm pretty sure we have non-Expo (in any way) projects that are in monorepos, and could provide guidance for those as well; this could be a todo for later though.

@jamonholmgren I think we might need to clarify intended scope, then. My impression is we were wanting to document monorepo configuration for future Ignite projects, and that's why this going in the Ignite docs. If we want a more general-purpose guide, are the Ignite docs the right place for that?

Reply:
Yeah, those could be a future Cookbook recipe @lindboe; we could link to them from here once they're published. It's always good to provide people with references to options, especially since those people might be us in the future!